samcli/lib/list/mapper_consumer_factory.py

```diff
@@ -4,14 +4,48 @@
 from samcli.lib.list.list_interfaces import MapperConsumerFactoryInterface
 from samcli.lib.list.data_to_json_mapper import DataToJsonMapper
 from samcli.commands.list.json_consumer import StringConsumerJsonOutput
+from samcli.commands.list.table_consumer import StringConsumerTableOutput
 from samcli.lib.list.mapper_consumer_container import MapperConsumerContainer
-from samcli.lib.list.list_interfaces import ProducersEnum
+from samcli.lib.list.stack_outputs.stack_output_to_table_mapper import StackOutputToTableMapper
+from samcli.lib.list.resources.resources_to_table_mapper import ResourcesToTableMapper
+from samcli.lib.list.testable_resources.testable_resources_to_table_mapper import TestableResourcesToTableMapper
+from samcli.lib.list.list_interfaces import ProducersEnum, Mapper
 
 
 class MapperConsumerFactory(MapperConsumerFactoryInterface):
+    """
+    Factory class to create factory objects that map a given producer and output format to a mapper and a consumer
+    """
+
     def create(self, producer: ProducersEnum, output: str) -> MapperConsumerContainer:
-        # Will add conditions here to return different sorts of containers later on
-        data_to_json_mapper = DataToJsonMapper()
-        json_consumer = StringConsumerJsonOutput()
-        container = MapperConsumerContainer(data_to_json_mapper, json_consumer)
+        """
+        Creates a MapperConsumerContainer that contains the resulting mapper and consumer given
+        the producer and output format
+
+        Parameters
+        ----------
+        producer: ProducersEnum
+            An enum representing the producers (stack-outputs, resources, or testable-resources producer)
+        output: str
+            The output format, either json or table
+
+        Returns
+        -------
+        container: MapperConsumerContainer
+            A MapperConsumerContainer containing the resulting mapper and consumer to be used by the producer
+        """
+        if output == "json":
```

Comment:
Something to think about: could we have this more generally across SAM CLI commands, instead of re-implementing for each? Similar to how AWS CLI has [`--output`](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-output-format.html) to specify the format for any time.

This would help tremendously in providing a consistent experience, and many commands would improve with parsable outputs (e.g. `sam local`, `sam deploy --no-deploy-changeset`, etc.).